deepchem/utils/openvino_model.py

```diff
@@ -71,27 +98,57 @@ def __init__(self,
     self._batch_size = batch_size
     self._outputs: List = []
 
+    self.input_size: List = []
+    self.output_size = None
+    self.last_batch_size = None
+    self.net = None
+
+    self.function = None
+
   def _read_torch_model(self, generator: Iterable[Tuple[Any, Any, Any]]):
     """
     Prepare PyTorch model for OpenVINO inference:
       1. Convert into ONNX format
       2. Read network from memory
     NOTE: We do not load model in __init__ method because of training.
     """
-    import torch
+    # import dgl
+    import torch.nn.functional as F
 
     # We need to serialize ONNX model with real input shape.
     # So we create a copy of the generator to take first input.
     generator_copy, generator = itertools.tee(generator, 2)
+
+    # list with array of objects deepchem.feat.graph_data.GraphData (len 4)
     inputs, _, _ = next(generator_copy)
-    assert (len(inputs) == 1), 'Not implemented'
-    inp_shape = list(inputs[0].shape)
-    inp_shape[0] = self._batch_size
+
+    if (type(self._torch_model).__name__ == 'CGCNNModel'):
+      # Create Graph object
+      inputs, _, _ = self._torch_model._prepare_batch((inputs, None, None))
+      self._torch_model.model.graph = inputs
+
+      node_feats = inputs.ndata.pop('x')
+      edge_feats = inputs.edata.pop('edge_attr')
+      inp = [node_feats, edge_feats]
+    else:
+      assert (len(inputs) == 1), 'Not implemented'
+
+      inp_shape = list(inputs[0].shape)
+      inp_shape[0] = self._batch_size
```

Comment:
Do we really need this step? I mean what is a shape of `inputs` for old models in case of batch_size > 1?

Reply:
@likholat, please take a look. We don't need to track batch size anymore